client/src/pages/Auth/SignIn.tsx

```diff
@@ -1,51 +1,168 @@
 import React from 'react';
-import { withRouter } from 'react-router-dom';
+import { connect } from "react-redux";
 
-class SignIn extends React.Component {
-  constructor(props:any) {
-    super(props);
+import {
+  ContainedButton,
+  PageBody,
+  WhiteTextTypography,
+  SecondaryMainTextTypography,
+  GreyBackgroundTextTypography,
+  Link
+} from "../../components/index";
 
-    const {
-      classes,
-      history,
-      user
-    } = props;
+import Typography from "@material-ui/core/Typography";
+import Grid from "@material-ui/core/Grid";
+import { OutlinedTextField } from "../../components/TextField";
+import InputAdornment from "@material-ui/core/InputAdornment";
 
-    if (user) {
-        // whatever the redirect route is supposed to be
-        history.push('/');
-    }
+/* Services */
+import { loginService } from "../../data/services/authServices";
+
+/* Selectors */
+import { getUser } from "../../data/selectors/userSelector";
+
+/* Type */
+import { User } from "../../data/types/userTypes";
+
+const mapStateToProps = (state: any): any => ({
+  user: getUser(state)
+});
+
+const mapDispatchToProps = (dispatch: any) => ({
+  login: (email: string, password: string) =>
+    dispatch(loginService(email, password)),
+});
+
+class SignIn extends React.Component<
+    { login: any, user: User, history: any }, 
+    { password: string; email: string; }
+  > {
+  constructor(props:any) {
+    super(props);
 
     this.state = {
       email: '',
       password: ''
     };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
   }
 
-  handleChange = () => {
+  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.id === "email") {
+      this.setState({email: e.target.value});
+    } else if (e.target.id === "password") {
+      this.setState({password: e.target.value});
+    }
```

Comment:
Really small, but I think you can simplify this to:

```suggestion
    }   handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { id, value } = event.target;
    this.setState({ ...this.state, [id]: value });
  };
```

Reply:
good catch